### DISCUSSION TITLE
`ColumnDef`: specifying `COLLATE utf8mb4_bin` (MySQL), `COLLATE BINARY` (SQLite) or `COLLATE "C"` (PostgreSQL)

### DISCUSSION THREAD

**Discussion by tp971** [SeaQL/sea-orm#1571] [Ideas]
I have a schema that needs certain columns to have a case sensitive (essentially byte-wise) collation. I imagine abstracting collations in general might be quite challenging since they are quite different between different DBMS, but it seems like most DBMS have some sort of byte-wise collation as mentioned in the title. I suggest this as an addition to at least `ColumnDef`, allowing to write migrations that specify byte-wise collations on certain columns.

**Comment by tyt2y3** [comment]
There is some support for MySQL in sea-query https://docs.rs/sea-query/latest/sea_query/table/struct.TableCreateStatement.html#method.collate

I am not sure what needs to be done for Postgres and SQLite, feel free to submit a proposal or PR.

**Reply by beautyfree** [reply]
https://github.com/SeaQL/sea-orm/issues/1901
